src/main/entity/Promotion.ts

```diff
@@ -82,13 +73,19 @@ export class Promotion {
   discount: Discount;
 
   /*
-   * OneToOne bidirectional relationship between Promotion and Restaurant
+   * ManyToOne bidirectional relationship between Promotion and Restaurant
    * Each promotion can have a single restaurant
+   * * IMPORTANT: We want delete the restaurant when restaurant no longer references any promotions.
+   * See RestaurantDeletion migration for how we handle this
+   *
+   * * We do NOT want cascade delete on foreign key restaurantId because we do not want to delete the promotions when a restaurant is deleted
+   * * Promotion is owning side of this association, contains column restaurantId
    * */
-  @OneToOne(() => Restaurant, (restaurant) => restaurant.promotion, {
-    cascade: true,
+  @ManyToOne(() => Restaurant, (restaurant) => restaurant.promotion, {
     nullable: false,
+    cascade: true,
```

Comment:
Just wanna double check- I thought we didn't want cascade delete?

Reply:
Ah yup this is tricky if you think of `cascade` in terms of cascade deletes. Actually, the cascade keyword for typeorm means if I have a promotion with a restaurant, and I save the promotion, the restaurant will be saved automatically. If I didn't have this keyword, I would need to save promotion first, then save restaurant separately